src/ExpressionEvaluator/Core/Source/ResultProvider/Expansion/DynamicViewExpansion.cs

```diff
@@ -0,0 +1,107 @@
+
+        internal static DynamicViewExpansion CreateExpansion(DkmInspectionContext inspectionContext, DkmClrValue value, Formatter formatter)
+        {
+            var type = value.Type.GetLmrType();
+            if (!(type.IsComObject() || type.IsIDynamicMetaObjectProvider()))
+            {
+                return null;
+            }
+
+            var proxyValue = value.InstantiateDynamicViewProxy(inspectionContext);
+            // InstantiateDynamicViewProxy may return null (if required assembly is missing, for instance).
+            if (proxyValue == null)
+            {
+                return null;
+            }
+
+            // Expansion is based on the 'DynamicMetaObjectProviderDebugView.Items' property.
+            var proxyType = proxyValue.Type;
+            var itemsMemberExpansion = RootHiddenExpansion.CreateExpansion(
+                proxyType.GetMemberByName("Items"),
+                DynamicFlagsMap.Create(new TypeAndCustomInfo(proxyType)));
+            return new DynamicViewExpansion(proxyValue, itemsMemberExpansion);
+        }
+
+        internal static EvalResultDataItem CreateMembersOnlyRow(
+            DkmInspectionContext inspectionContext,
+            string name,
+            DkmClrValue value,
+            Formatter formatter)
+        {
+            var expansion = CreateExpansion(inspectionContext, value, formatter);
+            return (expansion != null) ?
+                expansion.CreateDynamicViewRow(inspectionContext, name, parent: null, formatter: formatter) :
+                new EvalResultDataItem(name, Resources.DynamicViewNotDynamic);
+        }
+
+        private readonly DkmClrValue _proxyValue;
+        private readonly Expansion _proxyMembers;
+
+        private DynamicViewExpansion(DkmClrValue proxyValue, Expansion proxyMembers)
+        {
+            Debug.Assert(proxyValue != null);
+            Debug.Assert(proxyMembers != null);
+
+            _proxyValue = proxyValue;
+            _proxyMembers = proxyMembers;
+        }
+
+        internal override void GetRows(
+            ResultProvider resultProvider,
+            ArrayBuilder<EvalResultDataItem> rows,
+            DkmInspectionContext inspectionContext,
+            EvalResultDataItem parent,
+            DkmClrValue value,
+            int startIndex,
+            int count,
+            bool visitAll,
+            ref int index)
+        {
+            if (InRange(startIndex, count, index))
+            {
+                rows.Add(CreateDynamicViewRow(inspectionContext, Resources.DynamicView, parent, resultProvider.Formatter));
+            }
+
+            index++;
+        }
+
+        private EvalResultDataItem CreateDynamicViewRow(DkmInspectionContext inspectionContext, string name, EvalResultDataItem parent, Formatter formatter)
+        {
+            var proxyTypeAndInfo = new TypeAndCustomInfo(_proxyValue.Type);
+            var fullName = parent?.ChildFullNamePrefix ?? name;
```

Comment:
Fixed and added an Assert that name is not "Dynamic View" if we're assigning name to fullName (parent == null).